### DISCUSSION TITLE
Column names mis-match when loading certain columns

### DISCUSSION THREAD

**Discussion by Ossifragus** [xiaodaigh/JDF.jl#78] [General]
The relative order of the loaded columns seem to be always the same of the relative order the whole data (Is the order specified in the metadata?). For example with a `data` consisting two columns `"col1"` and `"col2"`, the code `JDF.load(data; cols = ["col1", "col2"])` loads the data correctly, but with `JDF.load(data; cols = ["col2", "col1"])` the data columns and their names would not match. 

Please see the following example:
```julia
using RDatasets, JDF, DataFrames

JDF.save("iris.jdf", dataset("datasets", "iris"))
a = DataFrame(JDF.load("iris.jdf"))

c1 = ["Species", "PetalWidth"]
a1 = DataFrame(JDF.load("iris.jdf"; cols = c1)) # column mis-match

julia> first(a1, 5)
5×2 DataFrame
 Row │ Species  PetalWidth 
     │ Float64  Cat…       
─────┼─────────────────────
   1 │     0.2  setosa
   2 │     0.2  setosa
   3 │     0.2  setosa
   4 │     0.2  setosa
   5 │     0.2  setosa


c2 = ["PetalWidth", "Species"]
a2 = DataFrame(JDF.load("iris.jdf"; cols = c2)) # column match

julia> first(a2, 5)
5×2 DataFrame
 Row │ PetalWidth  Species 
     │ Float64     Cat…    
─────┼─────────────────────
   1 │        0.2  setosa
   2 │        0.2  setosa
   3 │        0.2  setosa
   4 │        0.2  setosa
   5 │        0.2  setosa
```

**Comment by xiaodaigh** [comment]
columns are loaded as specified in the order by `cols`. This is the intended behaviour though

**Reply by Ossifragus** [reply]
> columns are loaded as specified in the order by `cols`. This is the intended behaviour though

So the first case in my example is indeed a bug, because the correct result should that the first column is the categorical variable, right?

**Reply by Ossifragus** [reply]
> columns are loaded as specified in the order by `cols`. This is the intended behaviour though

What do you mean? For `cols`, do you mean the order in the original full DataFrame? This may cause a lot of problems.

**Reply by xiaodaigh** [reply]
you very right. i missed that detail.

It's been fixed with the latest version of v0.5

**Reply by Ossifragus** [reply]
Great! Thank you so much!

**Reply by 4db83** [reply]
Hi, I still have the same problem with this. That is, the cols ordering is not satisfied. See code snipped below (Julia v1.7):

(@v1.7) pkg> status JDF
      Status `C:\Users\ibm\.julia\environments\v1.7\Project.toml`
  [babc3d20] JDF v0.5.0

julia> first(DataFrame(JDF.load("gkx-msf-merged.jdf", cols = [:date, :permno])),5)
5×2 DataFrame
 Row │ permno  date     
     │ Int64   Int64    
─────┼──────────────────
   1 │  10000  19860228 
   2 │  10000  19860331 
   3 │  10000  19860430 
   4 │  10000  19860530 
   5 │  10000  19860630 

julia> first(DataFrame(JDF.load("gkx-msf-merged.jdf", cols = [:permno, :date])),5)
5×2 DataFrame
 Row │ permno  date     
     │ Int64   Int64    
─────┼──────────────────
   1 │  10000  19860228 
   2 │  10000  19860331 
   3 │  10000  19860430 
   4 │  10000  19860530 
   5 │  10000  19860630 

julia>

**Reply by xiaodaigh** [reply]
I run the load in parallel so it's hard to force the column order. let me see

**Reply by xiaodaigh** [reply]
fixed by #81

**Comment by 4db83** [comment]
Thank you
